File: SystemGym.Service/CombosListService.cs
using System.Collections.Generic;
using System.Linq;
using Microsoft.EntityFrameworkCore;
using SystemGym.DataAccess.Models;
using SystemGym.Model.Address;
using SystemGym.Model.Ano;
using SystemGym.Model.FormaPagamento;
using SystemGym.Model.Funcao;
using SystemGym.Model.Mes;
using SystemGym.Model.Permissao;
using SystemGym.Model.Plano;
using SystemGym.Model.Sexo;
using SystemGym.Model.SituacaoAluno;
using SystemGym.Model.SituacaoColaborador;
using SystemGym.Model.SituacaoMatricula;
using SystemGym.Model.TipoNotificacao;

namespace SystemGym.Service
{
    public class CombosListService
    {
        private readonly SystemGymContext context;

        public CombosListService(SystemGymContext context)
        {
            this.context = context;
        }

        public List<SexoReturnModel> GetSexo()
        {
            return this.context.Sexo
               .ToList()
               .Select(x => new SexoReturnModel() {
                   SexoId = x.SexoId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }
        public List<PermissaoReturnModel> GetPermissao()
        {
            return this.context.Permissao
               .ToList()
               .Select(x => new PermissaoReturnModel()
               {
                   PermissaoId = x.PermissaoId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<SituacaoColaboradorReturnModel> GetSituacaoColaborador()
        {
            return this.context.SituacaoColaborador
               .ToList()
               .Select(x => new SituacaoColaboradorReturnModel()
               {
                   SituacaoColaboradorId = x.SituacaoColaboradorId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<SituacaoMatriculaReturnModel> GetSituacaoMatricula()
        {
            return this.context.SituacaoMatricula
               .ToList()
               .Select(x => new SituacaoMatriculaReturnModel()
               {
                   SituacaoMatriculaId = x.SituacaoMatriculaId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<TipoNotificacaoReturnModel> GetTipoNotificacao()
        {
            return this.context.TipoNotificacao
               .ToList()
               .Select(x => new TipoNotificacaoReturnModel()
               {
                   TipoNotificacaoId = x.TipoNotificacaoId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<PlanoReturnModel> GetPlano()
        {
            return this.context.Plano
               .ToList()
               .Select(x => new PlanoReturnModel()
               {
                   PlanoId = x.PlanoId,
                   Descricao = x.Descricao,
                   Valor = x.Valor,
               })
               .ToList();
        }

        public List<FuncaoReturnModel> GetFuncao()
        {
            return this.context.Funcao
               .ToList()
               .Select(x => new FuncaoReturnModel()
               {
                   FuncaoId = x.FuncaoId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<MesReturnModel> GetMes()
        {
            return this.context.Mes
               .ToList()
               .Select(x => new MesReturnModel()
               {
                   MesId = x.MesId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<AnoReturnModel> GetAno()
        {
            return this.context.Ano
               .ToList()
               .Select(x => new AnoReturnModel()
               {
                   AnoId = x.AnoId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

        public List<FormaPagamentoReturnModel> GetFormaPagamento()
        {
            return this.context.FormaPagamento
               .ToList()
               .Select(x => new FormaPagamentoReturnModel()
               {
                   FormaPagamentoId = x.FormaPagamentoId,
                   Descricao = x.Descricao,
               })
               .ToList();
        }

    }
}

File: SystemGym.Viewer/Angular/src/app/services/combosList.service.ts
import { Injectable } from '@angular/core';
import { HttpClient} from "@angular/common/http";

import * as moment from 'moment';
import { CustomErrorHandler } from '../core';
import { Sexo } from '../models/sexo-model';
import { Permissao } from '../models/permissao-model';
import { SituacaoColaborador } from '../models/situacao-colaborador-model';
import { SituacaoMatricula } from '../models/situacao-matricula-model';
import { TipoNotificacao } from '../models/tipo-notificacao-model';
import { Plano } from '../models/plano-model';
import { Funcao } from '../models/funcao-model';
import { Mes } from '../models/mes-model';
import { Ano } from '../models/ano-model';
import { FormaPagamento } from '../models/forma-pagamento-model';

@Injectable()
export class CombosListService {

  private endpointUrl = 'https://localhost:44365/api/CombosList';  // URL to web API

  constructor(private http: HttpClient) {
  }

  getSexo(): Promise<Sexo[]> {
    return this.http.get<Sexo[]>(this.endpointUrl + '/Sexo')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getPermissao(): Promise<Permissao[]> {
    return this.http.get<Permissao[]>(this.endpointUrl + '/Permissao')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getSituacaoColaborador(): Promise<SituacaoColaborador[]> {
    return this.http.get<SituacaoColaborador[]>(this.endpointUrl + '/SituacaoColaborador')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getSituacaoMatricula(): Promise<SituacaoMatricula[]> {
    return this.http.get<SituacaoMatricula[]>(this.endpointUrl + '/SituacaoMatricula')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getTipoNotificacao(): Promise<TipoNotificacao[]> {
    return this.http.get<TipoNotificacao[]>(this.endpointUrl + '/TipoNotificacao')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getPlano(): Promise<Plano[]> {
    return this.http.get<Plano[]>(this.endpointUrl + '/Plano')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getFuncao(): Promise<Funcao[]> {
    return this.http.get<Funcao[]>(this.endpointUrl + '/Funcao')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getMes(): Promise<Mes[]> {
    return this.http.get<Mes[]>(this.endpointUrl + '/Mes')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getAno(): Promise<Ano[]> {
    return this.http.get<Ano[]>(this.endpointUrl + '/Ano')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  getFormaPagamento(): Promise<FormaPagamento[]> {
    return this.http.get<FormaPagamento[]>(this.endpointUrl + '/FormaPagamento')
      .toPromise()
      .then(data => { return data; })
      .catch(this.handleError);
  }

  private handleError(error: any) {
    let errMsg = CustomErrorHandler.extractMessage(error);
    return Promise.reject(errMsg);
  }
}
